src/ngraph/pass/nop_elimination.cpp

```diff
@@ -133,7 +133,7 @@ bool ngraph::pass::NopElimination::run_on_function(std::shared_ptr<ngraph::Funct
 
         // Here we're checking on a common base class of a family of template classes,
         // which is more than type info can handle.
-        auto sclb = std::dynamic_pointer_cast<ngraph::op::ScalarConstantLikeBase>(n);
+        auto sclb = std::static_pointer_cast<ngraph::op::ScalarConstantLikeBase>(n);
```

Comment:
I think this one still needs to be `dynamic_pointer_cast`

Reply:
agreed. please put back to dynamic

Reply:
oops ... my bad